server/src/main/java/org/opensearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -642,7 +642,7 @@ public void handleException(TransportException exp) {
                         primaryRequest.getPrimaryTerm(),
                         initialRetryBackoffBound,
                         retryTimeout,
-                        indexShard.isRemoteTranslogEnabled()
+                        indexShard.isRemoteTranslogEnabled() || indexShard.indexSettings().isRemoteNode()
```

Comment:
Lets subsume the check of `indexShard.indexSettings().isRemoteNode()` inside the `indexShard.isRemoteTranslogEnabled()` check itself?